lib/SimplifyRust.ml

```diff
@@ -2,13 +2,86 @@
 
 open Ast
 
+(* Simplifications occuring on Ast *)
+
 (* Doesn't make sense in Rust *)
 let remove_push_pop_frame = object
   inherit [_] map
   method! visit_EPushFrame _ = EUnit
   method! visit_EPopFrame _ = EUnit
 end
 
-let simplify files =
+let simplify_ast files =
   let files = remove_push_pop_frame#visit_files () files in
   files
+
+
+(* Transformations occuring on the MiniRust AST, after translation and mutability inference *)
```

Comment:
do you mind putting those in OptimizeMiniRust? this is the file that has optimizations, for Rust, over the MiniRust AST, while SimplifyRust has optimizations, for Rust, on the Ast AST.

Reply:
Done in [9465c7d](https://github.com/FStarLang/karamel/pull/462/commits/9465c7d566ab7c55b0c9491888dfb43f7c5acc11)